test/files/pos/auto-application.scala

```diff
@@ -0,0 +1,31 @@
+// scalac: -Werror -deprecation -Xsource:2.14
+//
+class Test {
+  def a1(xs: List[String]): Int = xs.hashCode
+  def a2(xs: List[String]): Int = xs.hashCode()
+  def a3(xs: List[String]): String = xs.toString
+  def a4(xs: List[String]): String = xs.toString()
+  def a5(xs: List[String]): Class[_] = xs.getClass
+  def a6(xs: List[String]): Class[_] = xs.getClass()
+  def a7(xs: List[String]): Int = xs.##
+  def a8(xs: List[String]): Int = xs.##()
+  def a9(x: Address): String = x.toString
+  def a10(x: Address): String = x.toString()
+  def a11(x: A): String = x.toString
+  def a12(x: A): String = x.toString()
+  def a13(x: B): String = x.toString
+  def a14(x: B): String = x.toString()
+}
+
+case class Address()
+
+class A() {
+  override def toString(): String = "A()"
+}
+
+class B() {
+  override def toString: String = "B()"
+}
+
+case class C(c: Int) extends AnyVal
```

Comment:
Value class generates `this.underlying.hashCode`.

Reply:
Another test:
```
$ skala -Xsource:2.14
Welcome to Scala 2.13.1 (OpenJDK 64-Bit Server VM 11.0.3)

scala> class C(val i: Int) extends AnyVal { override def toString = "hi" }
                                                                    ^
       warning: Auto-application to `()` is deprecated. Supply the empty argument list `()` explicitly to invoke method toString$extension,
       or remove the empty argument list from its definition (Java-defined methods are exempt).
       In Scala 3, an unapplied method like this will be eta-expanded into a function.
defined class C

scala> class C(val i: Int) extends AnyVal { override def toString() = "hi" }
defined class C

```
My diff:
```
-          if (sourceLevel2_14 && !meth.isJavaDefined)
+          def isJavaOrUniversal(sym: Symbol) = sym.isJavaDefined || isUniversalMember(sym)
+          if (sourceLevel2_14 && !isJavaOrUniversal(meth) && !meth.overrides.exists(isJavaOrUniversal))
```